.github/workflows/dotnet-test.yml

```diff
@@ -43,8 +37,39 @@ jobs:
 
       - name: Test
         run: dotnet test --no-build --verbosity normal
-      
-      - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v2
-        with:
-          category: "/language:csharp"
+
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+      statuses: write
+    strategy:
+      matrix:
+          os: [ubuntu-latest, macos-latest, windows-latest]
+          language: [ 'csharp' ]
```

Comment:
Discussed offline, move language out of the Matrix params, since we only want jobs for each OS, but each OS job, can run for multiple languages (if we need more). Can be hardcoded to csharp for now in the analyze task.